go/geth_adapter/adapter_test.go

```diff
@@ -75,3 +86,311 @@ func TestRunContextAdapter_SetBalanceHasCorrectEffect(t *testing.T) {
 		})
 	}
 }
+
+func TestRunContextAdapter_SetAndGetNonce(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	stateDb := NewMockStateDb(ctrl)
+	adapter := &runContextAdapter{evm: &geth.EVM{StateDB: stateDb}}
+
+	address := tosca.Address{0x42}
+	nonce := uint64(123)
+
+	stateDb.EXPECT().SetNonce(gc.Address(address), nonce)
+	adapter.SetNonce(address, nonce)
+
+	stateDb.EXPECT().GetNonce(gc.Address(address)).Return(nonce)
+	adapter.GetNonce(address)
```

Comment:
nit: check that the `nonce` returned by the mock is also returned by the adapter